src/Integration.Vsix/Settings/SonarLintSettings.cs

```diff
@@ -23,44 +23,31 @@
 using Microsoft.VisualStudio.Settings;
 using Microsoft.VisualStudio.Shell;
 using Microsoft.VisualStudio.Shell.Interop;
-using Microsoft.VisualStudio.Shell.Settings;
+using SonarLint.VisualStudio.Integration.Vsix.Settings;
 
 namespace SonarLint.VisualStudio.Integration.Vsix
 {
     [Export(typeof(ISonarLintSettings))]
     [PartCreationPolicy(CreationPolicy.Shared)]
-    public class SonarLintSettings : ISonarLintSettings, IProfileManager
+    internal class SonarLintSettings : ISonarLintSettings, IProfileManager
     {
         public const string SettingsRoot = "SonarLintForVisualStudio";
 
-        private readonly WritableSettingsStore writableSettingsStore;
+        // Lazily create the settings store on first use (we can't create it in the constructor)
+        private readonly Lazy<WritableSettingsStore> writableSettingsStore;
 
         [ImportingConstructor]
-        public SonarLintSettings(SVsServiceProvider serviceProvider)
-            : this(new ShellSettingsManager(serviceProvider))
+        public SonarLintSettings(IWritableSettingsStoreFactory storeFactory)
         {
-        }
-
-        internal SonarLintSettings(SettingsManager settingsManager)
-        {
-            if (settingsManager == null)
-            {
-                throw new ArgumentNullException(nameof(settingsManager));
-            }
-
-            this.writableSettingsStore = settingsManager.GetWritableSettingsStore(SettingsScope.UserSettings);
-            if (this.writableSettingsStore != null &&
-                !this.writableSettingsStore.CollectionExists(SettingsRoot))
-            {
-                this.writableSettingsStore.CreateCollection(SettingsRoot);
-            }
+            // Called from MEF constructor -> must be free-threaded
+            writableSettingsStore = new Lazy<WritableSettingsStore>(() => storeFactory.Create(SettingsRoot));
         }
 
         internal /* testing purposes */ bool GetValueOrDefault(string key, bool defaultValue)
         {
             try
             {
-                return this.writableSettingsStore?.GetBoolean(SettingsRoot, key, defaultValue)
+                return writableSettingsStore.Value?.GetBoolean(SettingsRoot, key, defaultValue)
```

Comment:
All of the references in this class to `writableSettingsStore` have changed to `writableSettingsStore.Value`. There's a unit test that checks that the factory is only invoked once.